projects/aca-playwright-shared/src/api/queries-api.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Utils } from '../utils';
+import { ApiClientFactory } from './api-client-factory';
+
+export class QueriesApi {
+  private apiService: ApiClientFactory;
+
+  constructor() {
+    this.apiService = new ApiClientFactory();
+  }
+  static async initialize(userName: string, password?: string): Promise<QueriesApi> {
+    const classObj = new QueriesApi();
+    await classObj.apiService.setUpAcaBackend(userName, password);
+    return classObj;
+  }
+
+  async waitForSites(searchTerm: string, data: { expect: number }) {
+    try {
+      const sites = async () => {
+        const totalItems = await this.findSitesTotalItems(searchTerm);
+        if (totalItems !== data.expect) {
+          return Promise.reject(totalItems);
+        } else {
+          return Promise.resolve(totalItems);
+        }
+      };
+
+      return await Utils.retryCall(sites);
+    } catch (error) {
+      console.error(`QueriesApi waitForSites : catch : `);
+      console.error(`\tExpected: ${data.expect} items, but found ${error}`);
+    }
+  }
+
+  private async findSitesTotalItems(searchTerm: string): Promise<number> {
+    try {
+      const opts = {
```

Comment:
could you please set type for opts?

Reply:
@katarzynakita I see you have created your own type for this but if you check 'findSites' function declaration then you can see there is already existing type for opts (FindQuery). Could you please use this instead?
 
<img width="601" alt="image" src="https://github.com/Alfresco/alfresco-content-app/assets/115619721/978c5a71-8375-4bb7-9b62-63db0ee04113">